lib/decompress/zstd_decompress.c

```diff
@@ -72,6 +72,140 @@
 #endif
 
 
+
+/*************************************
+ * Multiple DDicts Hashset internals *
+ *************************************/
+
+#define DDICT_HASHSET_MAX_LOAD_FACTOR 0.75
+#define DDICT_HASHSET_TABLE_BASE_SIZE 64
+#define DDICT_HASHSET_RESIZE_FACTOR 2
+
+/* Hash function to determine starting position of dict insertion within the table
+ * Returns an index between [0, hashSet->ddictPtrTableSize]
+ */
+static size_t ZSTD_DDictHashSet_getIndex(const ZSTD_DDictHashSet* hashSet, U32 dictID) {
+    U32 hash = XXH64(&dictID, sizeof(U32), 0);
```

Comment:
One thing I can't quite figure out is why on certain builds/CI tests, `zstd_decompress.c` can't see `XXH64`, but can see `XXH64_digest(), etc.`? As far as I can tell, `xxhash.h` is within the graph of dependencies. Is there an additional macro I need to invoke?

Reply:
Is this related to the linux kernel test ?

Reply:
Yeah, I do see a line about `Replacing XXH64 prefix with xxh64` in the logs, and I assume it might be related.

Reply:
You should just need to change this line:

https://github.com/facebook/zstd/blob/bc0a1e4d59bc9541cb4740f08ae17d97a3fcc4b5/contrib/freestanding_lib/freestanding.py#L579

Something like: `re.compile(r"([^\w]|^)(?P<orig>XXH64)[\(_]")`